### PR TITLE
refactor v2

### PR DESCRIPTION
**Pull request by AleksyBalazinski**
Callback mechanism inspired by
https://www.gamedev.net/blogs/entry/2250186-designing-a-robust-input-handling-system-for-games/

### CODE REVIEW

**Review by KarolDenst — COMMENTED** [review]
Pressing middle mouse button throws exception. Same for all keys that don't do anything like 'g'.

**Review by KarolDenst — COMMENTED** [review]
Pressing middle mouse button throws exception. Same for all keys that don't do anything like 'g'.

File: Hyper/UserInput/Context.cs
Comment (KarolDenst) [inline Hyper/UserInput/Context.cs:1]: Pressing middle mouse button throws exception. Same for all keys that don't do anything like 'g'.

**Review by AleksyBalazinski — COMMENTED** [review]
FPS dropped significantly (4x on my pc) after merging

**Review by KarolDenst — COMMENTED** [review]
Projectiles don't move

File: Hyper/Meshes/Chunk.cs [hunk 1]
```diff
@@ -4,134 +4,129 @@
 using NLog;
```
Comment (KarolDenst) [inline Hyper/Meshes/Chunk.cs:4]: When moving if i press right mouse button i sometimes get a speed boost

Comment (KarolDenst) [inline Hyper/Meshes/Chunk.cs:4]: Chunk is only 1 color when there should be 3

**Review by KarolDenst — APPROVED** [review]

**Pull request merged**